examples/llm/src/models/mosaic_gpt.py

```diff
@@ -134,37 +134,66 @@ def _attn_bias(self, device, dtype):
     def forward(
             self,
             input_ids: torch.LongTensor,
-            key_padding_mask: Optional[torch.ByteTensor] = None,
-            past_key_values: Optional[List[Tuple[torch.FloatTensor]]] = None):
+            past_key_values: Optional[List[Tuple[torch.FloatTensor]]] = None,
+            attention_mask: Optional[torch.ByteTensor] = None,
+            return_dict: Optional[bool] = None,
+            output_attentions: Optional[bool] = None,
+            output_hidden_states: Optional[bool] = None,
+            use_cache: Optional[bool] = None):
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+        use_cache = use_cache if use_cache is not None else self.config.use_cache
+        if not return_dict:
+            raise NotImplementedError(
+                'return_dict False is not implemented yet for MosaicGPT')
+        if output_attentions:
+            raise NotImplementedError(
+                'output_attentions is not implemented yet for MosaicGPT')
+        if output_hidden_states:
+            raise NotImplementedError(
+                'output_hidden_states is not implemented yet for MosaicGPT')
+
+        if attention_mask is not None and attention_mask[:, 0].sum(
+        ) != attention_mask.shape[0] and self.training:
+            raise NotImplementedError(
+                'MosaicGPT does not support training with left padding.')
```

Comment:
Well, it might. I guess I'm not sure if the forward nans cause issues and I didn't test. Should I a) remove this check without trying it b) leave the check and file a jira to test it out later c) hold this PR to test it

Reply:
d) convert check into warning, file a jira to test it out later

Reply:
b) sounds good to me. We don't plan to train with left padding anytime soon, I think.

Reply:
I think I prefer the explicit error for now rather than a confusing failure. We can prioritize testing it out as needed, shouldn't take that long to try out and make sure convergence is preserved.